subprojects/core/src/main/java/org/gradle/api/execution/internal/ExecuteTaskBuildOperation.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.execution.internal;
+
+import org.gradle.api.Task;
+import org.gradle.api.internal.TaskInternal;
+import org.gradle.internal.operations.BuildOperationType;
+import org.gradle.internal.scan.UsedByScanPlugin;
+
+public final class ExecuteTaskBuildOperation implements BuildOperationType<ExecuteTaskBuildOperation.Details, Void> {
```

Comment:
All of these should be renamed to xxxBuildOperationType (if we want to go with that pattern). They are not `BuildOperation`s (a thing that gets executed), but metadata describing that operation.